src/Symfony/Component/Security/Http/EventListener/RememberMeListener.php

```diff
@@ -53,27 +71,61 @@ public function onSuccessfulLogin(LoginSuccessEvent $event): void
             return;
         }
 
-        if (null === $event->getResponse()) {
+        if (null === $response = $event->getResponse()) {
             if (null !== $this->logger) {
                 $this->logger->debug('Remember me skipped: the authenticator did not set a success response.', ['authenticator' => \get_class($event->getAuthenticator())]);
             }
 
             return;
         }
 
-        $this->rememberMeServices->loginSuccess($event->getRequest(), $event->getResponse(), $event->getAuthenticatedToken());
```

Comment:
This call was previously checking if the "remember me" flag/checkbox was set and therefore decided if the cookie is added or not. I'm wondering, where this logic would live after the rework. Can't find anything in the current changeset (I guess because it's WIP).

I'm especially interested in that aspect, because 2FA needs a way to prevent the default behavior (remember me cookie set right after login).

Reply:
Very good point, I completely forgot to implement all if statements in `AbstractRememberMeServices::loginSuccess()`